### ISSUE TITLE
Image from cv2.VideoCapture not working

### ISSUE THREAD

**Issue by Johndirr** [faustomorales/keras-ocr#25]
I don't know if this is a bug or not but when trying to run the `pipeline.recognize` on an image coming from a `cv2.VideoCapture` I get the following error:

> Traceback (most recent call last):
>   File "testing.py", line 106, in <module>
>     predictions = pipeline.recognize([newimg])[0]
>   File "C:\Users\RetroHelix\Envs\test\lib\site-packages\keras_ocr\pipeline.py", line 58, in recognize
>     **recognition_kwargs)
>   File "C:\Users\RetroHelix\Envs\test\lib\site-packages\keras_ocr\recognition.py", line 437, in recognize_from_boxes
>     for row in self.prediction_model.predict(X, **kwargs)
>   File "C:\Users\RetroHelix\Envs\test\lib\site-packages\tensorflow_core\python\keras\engine\training.py", line 909, in predict
>     use_multiprocessing=use_multiprocessing)
>   File "C:\Users\RetroHelix\Envs\test\lib\site-packages\tensorflow_core\python\keras\engine\training_arrays.py", line 715, in predict
>     x, check_steps=True, steps_name='steps', steps=steps)
>   File "C:\Users\RetroHelix\Envs\test\lib\site-packages\tensorflow_core\python\keras\engine\training.py", line 2472, in _standardize_user_data
>     exception_prefix='input')
>   File "C:\Users\RetroHelix\Envs\test\lib\site-packages\tensorflow_core\python\keras\engine\training_utils.py", line 565, in standardize_input_data
>     'with shape ' + str(data_shape))
> **ValueError: Error when checking input: expected input_2 to have 4 dimensions, but got array with shape (0, 1)**

The image I get from `cap.read` is a `numpy.ndarray` and the shape is the same as when using `keras_ocr.tools.read` so I think it should be ok to use. Any idea whats wrong?
Here is the relevant snippet of my code:

```
cap = cv2.VideoCapture(args.video)
while cap.isOpened():
    # read frame
    ret, frame = cap.read()
    print(frame.shape)
    print(type(frame))
    
    # predictions is a list of (text, box)
    predictions = pipeline.recognize([frame])[0]
```

**Comment by faustomorales** [comment]
It looks like you aren't checking for the video ending. So I suspect that this is working fine up until the point where the video ends, at which point `ret` will be `False`.

Immediately after `ret, frame = cap.read()`, you should check for the video being over by doing:

```python
if not ret:
    break
```

If this does not work, please do the following to help with debugging the issue:

1. Change `print(frame.shape)` to `print("Frame Shape:", frame.shape)`. This will make the logging easier to read.
2. Run your script and provide the 15 (or some other reasonable number) lines that are printed before the exception is raised, along with the traceback, as you have above.

**Comment by Johndirr** [comment]
The shape is fine: (480, 640, 3) and it's the first frame of the video. Thinking about it I think I know whats wrong. I can't test it right now but I think whats happening is, that the frame has no text at all and since I use `pipeline.recognize` it awaits a result after the detection to feed it to the recognizer. I was about to check if text was detected only after I get predictions.
Will test this in time.

**Comment by faustomorales** [comment]
You are 100% right. This happens whenever no text is detected in the image. I've pushed 5c1cb4f which fixes the issue and adds a test with a blank image to make sure this doesn't break again in the future. The fix is implemented in v0.6.2, which I've just published. Please try again and let me know if the issue persists.

**Comment by Johndirr** [comment]
Works like a charm now :)
